trantor/net/inner/tlsprovider/OpenSSLProvider.cc

```diff
@@ -714,13 +714,11 @@ struct OpenSSLProvider : public TLSProvider, public NonCopyable
     {
         constexpr size_t maxSingleRead = 128 * 1024;
         constexpr size_t maxWritibleBytes = (std::numeric_limits<int>::max)();
-        auto inBio = SSL_get_rbio(ssl_);
         while (true)
         {
-            auto pending = BIO_pending(inBio);
-            if (pending <= 0)
-                break;
+            auto pending = BIO_pending(rbio_);
             // horrible syntax, because MSVC
+            pending = (std::min)(1024, pending);
```

Comment:
Considering the case of pending==0, we need ensureWritableBytes to 1024 bytes, so here should be std::max.